src/MassTransit/Middleware/HandlerMessageFilter.cs

```diff
@@ -53,18 +54,15 @@ async Task IFilter<ConsumeContext<TMessage>>.Send(ConsumeContext<TMessage> conte
             }
             catch (OperationCanceledException exception)
             {
-                await context.NotifyFaulted(timer.Elapsed, TypeCache<MessageHandler<TMessage>>.ShortName, exception).ConfigureAwait(false);
-
-                if (exception.CancellationToken == context.CancellationToken)
-                    throw;
-
-                throw new ConsumerCanceledException($"The operation was canceled by the consumer: {TypeCache<MessageHandler<TMessage>>.ShortName}");
+                await ThrowHelper.ConsumeFilter.ThrowOperationCancelled<MessageHandler<TMessage>, TMessage>(context, timer.Elapsed, exception, activity)
+                    .ConfigureAwait(false);
             }
             catch (Exception ex)
             {
-                await context.NotifyFaulted(timer.Elapsed, TypeCache<MessageHandler<TMessage>>.ShortName, ex).ConfigureAwait(false);
-
                 Interlocked.Increment(ref _faulted);
+
+                await ThrowHelper.ConsumeFilter.Throw<MessageHandler<TMessage>, TMessage>(context, timer.Elapsed, ex, activity).ConfigureAwait(false);
```

Comment:
not really a fan of this style of helper class stuff.

Reply:
Was trying to avoid repeating it over and over again, any suggestions? If not I can just take it out

Reply:
I'd suggest removing it, it adds stack depth to the exception, vs. just using `throw` in the scope to rethrow the existing exception with the original call stack.